src/plugins/query_enhancements/public/query_assist/components/query_assist_summary.tsx

```diff
@@ -83,18 +89,23 @@ export const QueryAssistSummary: React.FC<QueryAssistSummaryProps> = (props) =>
 
   const sampleSize = 10;
 
-  const reportMetric = useCallback(
-    (metric: string) => {
-      if (props.usageCollection) {
-        props.usageCollection.reportUiStats(
-          METRIC_APP,
-          props.usageCollection.METRIC_TYPE.CLICK,
-          metric + '-' + uuidv4()
-        );
-      }
-    },
-    [props.usageCollection, METRIC_APP]
-  );
+  const queryChanged =
```

Comment:
Will this be recalculated whenever there's state change?

Reply:
Yes, the change of `queryState` will make the component rerender.